src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -728,6 +726,15 @@ void ClusterTaskManager::FillResourceUsage(rpc::ResourcesData &data) {
       by_shape_entry->set_backlog_size(backlog_it->second);
     }
   }
+
+  // Check whether resources have been changed.
+  auto last_heartbeat_resources = gcs_client_->NodeResources().GetLastResourceUsage();
```

Comment:
Can we save an RPC by having the resource report poller send this information instead of requesting it?

It's much easier to make GCS stable if we don't send requests to it.

Reply:
Yep it's critical we don't send rpcs directly to the GCS like this. Instead, the GCS should initiate rpcs so that resource polling gracefully degrades under overload.

Reply:
So to summarize:

1. maybe in short term we can reuse the last_heartbeat_resources in [NodeManager::FillResourceReport](https://github.com/ray-project/ray/blob/41cf2e3d50105a8bc42939e0bc32c6423bd95b55/src/ray/raylet/node_manager.cc#L557) and pass it in, which doesn't make thing worse.
2.  In longer term we need to change it from polling to pushing?

Reply:
Ideally we can do (2) now.

I was under the impression that Ant was already running into dropped messages, so (1) would cause a regression. 

(2) shouldn't be very complex to implement. You can add it to the existing `RequestResourceReportRequest` message.

Reply:
Actually `gcs_client_->NodeResources().GetLastResourceUsage()` doesn't send a RPC. It's just a getter method of a cache variable. For saving the dependency of the gcs_client, we can use the last_heartbeat_resources as @scv119 said.

BTW, we are not using resource report poller now. Is this the new thing you added recently? I'm wondering what is your plan. I have some questions:
1. I found that resource report poller is enabled by default now. Are pulling and pushing working together now?
2. Are you planning to migrate to resource report poller eventually? I'm concerned about the resource updating latency if we use polling instead of pushing.

Reply:
> I was under the impression that Ant was already running into dropped messages, so (1) would cause a regression.

@wuisawesome  I didn't understand this, why it's a regression? I think in this PR, let's just do (1) as @scv119 said, fix the problem in pushing. I can open another PR to do this for polling. Do you think it's OK?